src/KafkaNET.Library/Consumers/PartitionTopicInfo.cs

```diff
@@ -218,6 +220,39 @@ public long FetchOffset
         // Gets or sets the offset for the next request.
         public long NextRequestOffset { get; internal set; }
 
+        public bool ConsumedOffsetOutOfRange
+        {
+            get
+            {
+                lock(consumedOffsetOutOfRangeLock)
+                {
+                    return this.consumedOffsetOutOfRangeFlag;
+                }
+            }
+            internal set
+            {
+                lock (consumedOffsetOutOfRangeLock)
+                {
+                    this.consumedOffsetOutOfRangeFlag = value;
+
+                    if (this.consumedOffsetOutOfRangeFlag)
+                    {
+                        // Clear the chunkQueue as current messages in the queue are invalid on brokers. Drop them in order to fetch valid messages in.
+                        while (this.chunkQueue.Count > 0)
```

Comment:
Summarizing our conversation offline:
1) chunkQueue will get a max capacity to help it force a strong limit on its usage
2) emptying the chunkQueue code will be removed